graphblas/core/ss/dtypes.py

```diff
@@ -0,0 +1,88 @@
+import numpy as np
+
+from ... import backend, core, dtypes
+from ...exceptions import check_status_carg
+from .. import _has_numba, ffi, lib
+from . import _IS_SSGB7
+
+ffi_new = ffi.new
+if _IS_SSGB7:
+    # JIT was introduced in SuiteSparse:GraphBLAS 8.0
+    import suitesparse_graphblas as ssgb
+
+    raise ImportError(
+        "JIT was added to SuiteSparse:GraphBLAS in version 8; "
+        f"current version is {ssgb.__version__}"
+    )
+if _has_numba:
+    import numba
+    from cffi import FFI
+    from numba.core.typing import cffi_utils
+
+    jit_ffi = FFI()
+
+
+def register_new(name, jit_c_definition, *, np_type=None):
```

Comment:
Should we remove the `name` argument here? It's required to be the same name as the typedef, and I bet we could determine the name from `jit_c_definition`.